firefox_ui_tests/firefox_ui_tests/functional/security/test_safe_browsing_primary_download.py

```diff
@@ -0,0 +1,75 @@
+import os
+import sys
+
+from marionette_driver import By, Wait
+from firefox_puppeteer.testcases import FirefoxTestCase
+
+
+class TestSafeBrowsingPrimaryDownload(FirefoxTestCase):
+
+    data = [
+        {
+            'platform_list': ['linux2', 'win32', 'darwin'],
+            'files': [
+                # Phishing
+                "goog-badbinurl-shavar.cache",
+                "goog-badbinurl-shavar.pset",
+                "goog-badbinurl-shavar.sbstore",
+                "goog-malware-shavar.cache",
+                "goog-malware-shavar.pset",
+                "goog-malware-shavar.sbstore",
+                "goog-phish-shavar.cache",
+                "goog-phish-shavar.pset",
+                "goog-phish-shavar.sbstore",
+                "goog-unwanted-shavar.cache",
+                "goog-unwanted-shavar.pset",
+                "goog-unwanted-shavar.sbstore",
+
+                # Tracking Protections
+                "mozstd-track-digest256.cache",
+                "mozstd-track-digest256.pset",
+                "mozstd-track-digest256.sbstore",
+                "mozstd-trackwhite-digest256.cache",
+                "mozstd-trackwhite-digest256.pset",
+                "mozstd-trackwhite-digest256.sbstore"
+                ]
+            },
+        {
+            'platform_list': ['win32'],
+            'files': [
+                "goog-downloadwhite-digest256.cache",
+                "goog-downloadwhite-digest256.pset",
+                "goog-downloadwhite-digest256.sbstore"
+            ]
+        }
+    ]
+
+    browser_prefs = {
+        'browser.safebrowsing.downloads.enabled': 'true',
+        'browser.safebrowsing.downloads.remote.enabled': 'true',
+        'browser.safebrowsing.enabled': 'true',
+        'browser.safebrowsing.malware.enabled': 'true',
+        'browser.safebrowsing.provider.google.nextupdatetime': 1,
+        'browser.safebrowsing.provider.mozilla.nextupdatetime': 1,
+        'privacy.trackingprotection.pbmode.enabled': 'true'
+    }
+
+    def setUp(self):
+        FirefoxTestCase.setUp(self)
+
+        # Restart Browser
+        self.restart()
+
+        # Set Browser Preferences
+        for item, value in self.browser_prefs.items():
+            self.prefs.set_pref(item, value)
```

Comment:
We still want this before the restart otherwise it could cause race conditions. Also I haven't seen any comment from you regarding my question on IRC earlier this week, if the preferences as set here are getting reset with the restart. Please check that.